code/game/mecha/mecha_topic.dm

```diff
@@ -216,43 +216,48 @@
 	if(usr.incapacitated())
 		return
 
-	var/datum/topic_input/afilter = new /datum/topic_input(href,href_list)
-
 	if(in_range(src, usr))
+		var/mob/user
+		var/obj/item/card/id/id_card
+		if (href_list["user"])
```

Comment:
Maybe out of scope but why in the world is this part of the topic call, there is an exploit here to basically show popup dialogs to anyone you can get the REF of. Should just be `usr`.